src/FilterBundle/DependencyInjection/CompilerPass/TermNormalizerPass.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace BestIt\Commercetools\FilterBundle\DependencyInjection\CompilerPass;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Reference;
+
+/**
+ * Class TermNormalizerPass
+ *
+ * @author Michel Chowanski <chowanski@bestit-online.de>
+ * @package BestIt\Commercetools\FilterBundle\DependencyInjection\CompilerPass
+ */
+class TermNormalizerPass implements CompilerPassInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
```

Comment:
Ganz schön viele Schleifen hier drin ^^

Reply:
"Quasi Standard" bei CompilerPass. In neueren Symfony Versionen geht dies weitaus eleganter. Da wir aber auch für 3.1 Kompatibel sein müssen, machen wir es so wie für 3.1. Und da muss man leider von allen Services einmal alle Tags durchgehen, separat speichern, sortieren und die Definition mit den sortieren Daten füttern.

https://symfony.com/doc/current/service_container/tags.html